sdk/formrecognizer/perf-tests/test/custom.spec.ts

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { PerfStressOptionDictionary, PerfStressTest } from "@azure/test-utils-perfstress";
+import {
+  AzureKeyCredential,
+  BeginRecognizeCustomFormsOptions,
+  CustomFormModel,
+  FormRecognizerClient,
+  FormTrainingClient
+} from "@azure/ai-form-recognizer";
+import { TokenCredential } from "@azure/core-http";
+import { DefaultAzureCredential } from "@azure/identity";
+
+function unreachable(message?: string): never {
+  throw new Error(message ?? "Unreachable Exception.");
+}
+
+export class CustomModelRecognitionTest extends PerfStressTest<BeginRecognizeCustomFormsOptions> {
+  public options: PerfStressOptionDictionary<BeginRecognizeCustomFormsOptions> = {
+    updateIntervalInMs: {
+      required: false,
+      description: "Polling interval in milliseconds",
+      shortName: "u",
+      longName: "update-interval",
+      defaultValue: 5000
+    }
+  };
+
+  /**
+   * Not thrilled about this, but `globalSetup` only runs once overall, while `setup`
+   * shouldn't have to train the model every time.
+   */
+  static sessionModel: CustomFormModel | undefined = undefined;
+
+  private recognizerClient: FormRecognizerClient;
+  private trainingClient: FormTrainingClient;
+
+  private documentUrl: string;
+
+  constructor() {
+    super();
+
+    let credential: TokenCredential | AzureKeyCredential;
+
+    try {
+      credential = new DefaultAzureCredential();
+    } catch {
+      credential = new AzureKeyCredential(process.env.FORM_RECOGNIZER_API_KEY ?? "");
+    }
+
+    const endpoint = process.env.FORM_RECOGNIZER_ENDPOINT;
+
+    if (!endpoint) {
+      throw new Error(`Expected FORM_RECOGNIZER_ENDPOINT to be defined: ${endpoint}`);
+    }
```

Comment:
[NIT]
`perfstress` package provides `getEnvVar` method that throws if the variable is not defined.
```suggestion
    const endpoint = getEnvVar("FORM_RECOGNIZER_ENDPOINT");
```